pkg/csi_driver/multishare_controller.go

```diff
@@ -334,13 +339,18 @@ func (m *MultishareController) getShareAndGenerateCSIControllerExpandVolumeRespo
 	}, nil
 }
 
-func (m *MultishareController) waitOnWorkflow(ctx context.Context, workflow *Workflow) (err error) {
+func (m *MultishareController) waitOnWorkflow(ctx context.Context, workflow *Workflow) (error, *codes.Code) {
 	timeout, pollInterval, err := util.GetMultishareOpsTimeoutConfig(workflow.opType)
 	if err != nil {
-		return
+		code := codes.Internal
+		return err, &code
 	}
 	err = m.cloud.File.WaitForOpWithOpts(ctx, workflow.opName, file.PollOpts{Timeout: timeout, Interval: pollInterval})
```

Comment:
For the error returned in the poll operation, we should have a function for resource exhausted (the IsUserError() handles that), as we can potentially hit that. We do not want to mask those errors as deadline exceeded.

Reply:
So you are saying in WaitForOpWithOpts, this line
[pollOp, err := manager.multishareOperationsServices.Get(op).Context(ctx).Do()](
https://github.com/kubernetes-sigs/gcp-filestore-csi-driver/blob/ad542caf0e1276e49016b3c33735de4515e8e318/pkg/cloud_provider/file/file.go#L866) can return ResourceExhausted? Looking at the definition of Do(), it also wraps the errors it returns: https://github.com/kubernetes-sigs/gcp-filestore-csi-driver/blob/ad542caf0e1276e49016b3c33735de4515e8e318/vendor/google.golang.org/api/file/v1beta1/file-gen.go#L6490

Reply:
we can occasionally get 404 too, if the resource disappeared between poll interval I think

Reply:
Hm ok I so return Internal error code for anything returned from `GetMultishareOpsTimeoutConfig()` or [IsOpDone()](https://github.com/kubernetes-sigs/gcp-filestore-csi-driver/blob/ad542caf0e1276e49016b3c33735de4515e8e318/pkg/cloud_provider/file/file.go#L593). 

Then for  Do() we need to map 404,  resource exhausted, and deadline exceeded to the correct error code, then for all others we return an internal error code. The polling error that has the "context deadline exceeded" message isn't actually a googleapi error right? Should I just parse the error string to see if it has that string, if so return DeadlineExceeded, otherwise return Internal?

Reply:
Yes thats sounds a reasonable approach